src/runtime/dom-extras.ts

```diff
@@ -63,6 +63,55 @@ export const patchSlotAppendChild = (HostElementPrototype: any) => {
   };
 };
 
+/**
+ * Patches the text content of an unnamed slotted node inside a scoped component
+ * @param hostElementPrototype the `Element` to be patched
+ * @param cmpMeta component runtime metadata used to determine if the component should be patched or not
+ */
+export const patchTextContent = (hostElementPrototype: HTMLElement, cmpMeta: d.ComponentRuntimeMeta): void => {
+  if (BUILD.scoped && cmpMeta.$flags$ & CMP_FLAGS.scopedCssEncapsulation) {
+    const descriptor = Object.getOwnPropertyDescriptor(Node.prototype, 'textContent');
+
+    Object.defineProperty(hostElementPrototype, '__textContent', descriptor);
+
+    Object.defineProperty(hostElementPrototype, 'textContent', {
+      get(): string | null {
+        // get the 'default slot', which would be the first slot in a shadow tree (if we were using one), whose name is
+        // the empty string
+        const slotNode = getHostSlotNode(this.childNodes, '');
+        // when a slot node is found, the textContent will be found in the next sibling (text) node. only if we can
+        // find that text node should we try to pull a value from it
+        if (slotNode?.nextSibling) {
```

Comment:
Why would this need to access the next sibling? Using `cmp-label` as an example, I assume `slotNode` and `slotNode.nextSibling` are the following: 
```
<Host>
  <label>
    <slot /> <!-- slotNode -->
    <div>Non-slotted text</div> <!-- slotNode.nextSibling -->
  </label>
</Host>
```

Reply:
it's because when not using `shadow: true` stencil tries to recreate the behaviour of slots using empty text nodes to preserve the slot position. Thus the next nodes after the slot text node, should be some slotted content.

Reply:
@ltm not exactly. Let's say we have a layout that looks something like:
```
<Host>
  <label>
    <slot />
    <div>Non-slot</div>
   </label>
</Host>
```
When we're doing slot relocation, calling `childNodes` on the label element yields:
![Screen Shot 2021-09-08 at 1 59 45 PM](https://user-images.githubusercontent.com/1930213/132561814-bb705826-65e9-4705-909e-7316d43a0e59.png)
where `slotNode` is that first text node, and `slotNode.nextSibling` in this case is the second text node.

When we call `getHostSlotNode`, it finds the first text node, and assigns it to `slotNode` because that node has the Stencil marker '[s-sn]` (slot name) set to an empty string (which is what we wanted). 

However, in cases like this, the VDOM has placed the content in that _second_ text node. Attempting to place the value received when setting `textContent` in the first node will result in concatenation of that received value (from setting `textContent`) and what's in the second text node.  So, we check here for this case first.

However, you also presented a case internally where this isn't always the case. I've updated this PR with additonal tests and hopefully clearer comments. LMK what you think